Source/WebCore/Modules/webauthn/fido/DeviceRequestConverter.cpp

```diff
@@ -88,13 +89,41 @@ static CBORValue convertDescriptorToCBOR(const PublicKeyCredentialDescriptor& de
     return CBORValue(WTFMove(cborDescriptorMap));
 }
 
-Vector<uint8_t> encodeMakeCredentialRequestAsCBOR(const Vector<uint8_t>& hash, const PublicKeyCredentialCreationOptions& options, UVAvailability uvCapability, AuthenticatorSupportedOptions::ResidentKeyAvailability residentKeyAvailability, const Vector<String>& authenticatorSupportedExtensions, std::optional<PinParameters> pin)
+static Vector<PublicKeyCredentialParameters> trimmedParameters(const Vector<PublicKeyCredentialParameters>& parameters, const std::optional<Vector<WebCore::PublicKeyCredentialParameters>>& authenticatorSupportedParameters)
+{
+    HashSet<int64_t> authenticatorSupportedAlgorithms;
+    if (authenticatorSupportedParameters) {
+        for (auto& parameters : *authenticatorSupportedParameters) {
+            if (parameters.type == PublicKeyCredentialType::PublicKey)
+                authenticatorSupportedAlgorithms.add(parameters.alg);
+        }
+    }
+
+    for (auto& parameter : parameters) {
+        if (parameter.type != PublicKeyCredentialType::PublicKey)
+            continue;
+        if (authenticatorSupportedAlgorithms.contains(parameter.alg))
+            return { parameter };
```

Comment:
So this always returns the first option supported by the authenticator. Does that cause any problem if there is a "preferred" algorithm that happens to fall later in the list of choices?

Reply:
The list is ordered from most preferred to least preferred. 

> This sequence is ordered from most preferred (by the RP) to least preferred.
from https://fidoalliance.org/specs/fido-v2.0-ps-20190130/fido-client-to-authenticator-protocol-v2.0-ps-20190130.html